package/scripts/ambari_helpers.py

```diff
@@ -43,7 +43,20 @@ def cdap_config(name=None):
        content=InlineTemplate(params.cdap_env_sh_template)
   )
 
-  cleanup_opts()
+  if name == 'auth':
+    dirname = 'security'
+  elif name == 'router':
+    dirname = 'gateway'
+  else:
+    dirname = name
+
+  cleanup_opts(dirname)
+
+  # Copy logback.xml and logback-container.xml
+  for i in 'logback.xml', 'logback-container.xml':
+    no_op_test = 'ls ' + params.cdap_conf_dir + '/' + i + ' 2>/dev/null'
+    Execute('cp -f /etc/cdap/conf.dist/' + i + ' ' + params.cdap_conf_dir, not_if=no_op_test)
```

Comment:
It's more the making sure that all the spaces and delimiters are there...the trade-off is as you say. It's 
```
Execute('cp -f /etc/cdap/conf.dist/' + i + ' ' + params.cdap_conf_dir, not_if=no_op_test)
```
versus
```
Execute("cp -f /etc/cdap/conf.dist/%s %s" % (i, params.cdap_conf_dir), not_if=no_op_test)
```